client/src/editor.ts

```diff
@@ -0,0 +1,66 @@
+import { ExtensionContext, Range, SnippetString, TextDocument, commands, window } from "vscode";
+
+export function initialise(context: ExtensionContext) {
+	context.subscriptions.push(
+		commands.registerCommand("vscode-rpgle.surroundWithMonitor", () => surroundWithSnippet(lines => [
+			"",
+			"monitor;",
+			...lines,
+			"on-excp '$1';",
+			"on-error$2;",
+			"endmon;",
+			""
+		]))
+	);
+}
+
+function surroundWithSnippet(snippetProvider: (lines: string[]) => string[]) {
+	const editor = window.activeTextEditor;
+	if (editor) {
+		const document = editor.document;
+		let line = editor.selection.start.line;
+		if (isFreeRPGLE(document, line)) {
```

Comment:
This logic is interesting, checking if the logic is currently free format. Perhaps instead if the document is not `**free`, substring the first 8(?) characters and then surround that? That might be nicer than checking for `/free` and `/end-free`.